pkg/provider/provider.go

```diff
@@ -218,278 +158,184 @@ func (p *Provider) getVaultURL(ctx context.Context) (vaultURL *string, err error
-			clientID := nmiResp.ClientID
-
-			if token.AccessToken == "" || clientID == "" {
-				return nil, fmt.Errorf("nmi did not return expected values in response: token and clientid")
-			}
-
-			spt, err := adal.NewServicePrincipalTokenFromManualToken(*oauthConfig, clientID, resource, token, nil)
-			if err != nil {
-				return nil, err
-			}
-			return spt, nil
-		}
-
-		err = fmt.Errorf("nmi response failed with status code: %d", resp.StatusCode)
-		return nil, err
-	}
-
-	if p.UseVMManagedIdentity {
-		msiEndpoint, err := adal.GetMSIVMEndpoint()
-		if err != nil {
-			return nil, errors.Wrap(err, "failed to get managed identity (MSI) endpoint")
-		}
-
-		if p.UserAssignedIdentityID != "" {
-			log.Infof("azure: using user assigned managed identity %s to retrieve access token", RedactClientID(p.UserAssignedIdentityID))
-			return adal.NewServicePrincipalTokenFromMSIWithUserAssignedID(
-				msiEndpoint,
-				resource,
-				p.UserAssignedIdentityID)
-		}
-
-		log.Infof("azure: using system assigned managed identity to retrieve access token")
-		return adal.NewServicePrincipalTokenFromMSI(
-			msiEndpoint,
-			resource)
-	}
-
-	// When CSI driver is using a Service Principal clientid + client secret to retrieve token for resource
-	if len(p.AADClientSecret) > 0 {
-		log.Infof("azure: using client_id+client_secret to retrieve access token")
-		return adal.NewServicePrincipalToken(
-			*oauthConfig,
-			p.AADClientID,
-			p.AADClientSecret,
-			resource)
-	}
-	return nil, fmt.Errorf("No credentials provided for AAD application %s", p.AADClientID)
+	return p.AuthConfig.GetServicePrincipalToken(p.PodName, p.PodNamespace, resource, p.AzureCloudEnvironment.ActiveDirectoryEndpoint, p.TenantID)
 }
 
 // MountSecretsStoreObjectContent mounts content of the secrets store object to target path
-func (p *Provider) MountSecretsStoreObjectContent(ctx context.Context, attrib map[string]string, secrets map[string]string, targetPath string, permission os.FileMode) (err error) {
+func (p *Provider) MountSecretsStoreObjectContent(ctx context.Context, attrib map[string]string, secrets map[string]string, targetPath string, permission os.FileMode) (map[string]string, error) {
 	keyvaultName := attrib["keyvaultName"]
 	cloudName := attrib["cloudName"]
 	usePodIdentityStr := attrib["usePodIdentity"]
 	useVMManagedIdentityStr := attrib["useVMManagedIdentity"]
 	userAssignedIdentityID := attrib["userAssignedIdentityID"]
 	tenantID := attrib["tenantId"]
+	cloudEnvFileName := attrib["cloudEnvFileName"]
 	p.PodName = attrib["csi.storage.k8s.io/pod.name"]
 	p.PodNamespace = attrib["csi.storage.k8s.io/pod.namespace"]
-	cloudEnvFileName := attrib["cloudEnvFileName"]
 
 	if keyvaultName == "" {
-		return fmt.Errorf("keyvaultName is not set")
+		return nil, fmt.Errorf("keyvaultName is not set")
 	}
 	if tenantID == "" {
-		return fmt.Errorf("tenantId is not set")
+		return nil, fmt.Errorf("tenantId is not set")
+	}
+	if len(usePodIdentityStr) == 0 {
+		usePodIdentityStr = "false"
```

Comment:
if `usePodIdentityStr` is not used anywhere else, we should set `usePodIdentity` here

Reply:
This is being used - https://github.com/Azure/secrets-store-csi-driver-provider-azure/pull/203/files#diff-6155820fac2dd2fda7353505b1f69319R198-R209

Reply:
This is a nit so feel free to ignore it. 

I meant why not set `usePodIdentity` immediate following this? ie. move up https://github.com/Azure/secrets-store-csi-driver-provider-azure/blob/66c131910d8da0e9aadae02d7260fd30fb245077/pkg/provider/provider.go#L198-L210

Reply:
Yeah, that makes sense! Moved it up.